docs/usage/other_features/zeppelin_context.md

```diff
@@ -0,0 +1,230 @@
+myScalaDataFrame = ...
+z.put("myScalaDataFrame", myScalaDataFrame)
+
+val myPythonDataFrame = z.get("myPythonDataFrame").asInstanceOf[DataFrame]
+{% endhighlight %}
+
+  </div>
+  <div data-lang="python" markdown="1">
+
+{% highlight python %}
+# Get object from python
+%spark.pyspark
+myObject = z.get("objName")
+
+# Exchanging data frames
+myPythonDataFrame = ...
+z.put("myPythonDataFrame", postsDf._jdf)
+
+myScalaDataFrame = DataFrame(z.get("myScalaDataFrame"), sqlContext)
+{% endhighlight %}
+
+  </div>
+</div>
+
+### Form Creation
+
+`ZeppelinContext` provides functions for creating forms.
+In Scala and Python environments, you can create forms programmatically.
+<div class="codetabs">
+  <div data-lang="scala" markdown="1">
+
+{% highlight scala %}
+%spark
+/* Create text input form */
+z.input("formName")
+
+/* Create text input form with default value */
+z.input("formName", "defaultValue")
+
+/* Create select form */
+z.select("formName", Seq(("option1", "option1DisplayName"),
+                         ("option2", "option2DisplayName")))
+
+/* Create select form with default value*/
+z.select("formName", "option1", Seq(("option1", "option1DisplayName"),
+                                    ("option2", "option2DisplayName")))
+{% endhighlight %}
+
+  </div>
+  <div data-lang="python" markdown="1">
+
+{% highlight python %}
+%spark.pyspark
+# Create text input form
+z.input("formName")
+
+# Create text input form with default value
+z.input("formName", "defaultValue")
+
+# Create select form
+z.select("formName", [("option1", "option1DisplayName"),
+                      ("option2", "option2DisplayName")])
+
+# Create select form with default value
+z.select("formName", [("option1", "option1DisplayName"),
+                      ("option2", "option2DisplayName")], "option1")
+{% endhighlight %}
+
+  </div>
+</div>
+
+In sql environment, you can create form in simple template.
+
+```sql
+%spark.sql
+select * from ${table=defaultTableName} where text like '%${search}%'
+```
+
+To learn more about dynamic form, checkout [Dynamic Form](../usage/dynamic_form/intro.html).
+
+### Interpreter-Specific Functions
+
+Some interpreters use a subclass of `BaseZepplinContext` augmented with interpreter-specific
+functions. For example the dependency loader (%spark.dep) has `addRepo()`, `load()`, and other
+related methods that implement its functionality. Interpreter-specific functions are
+described within each interpreter's documentation.
+
+## Usage in Non-Programming Cells
+
+Cells with a non-programming interpreter (see table below) use a pattern match-and-replace 
+approach to use two of zeppelin-context's features: dynamic forms and 
+object interpolation. 
+
+|                   Non-Programming Interpreters                    |
+|-------------------------------------------------------------------|
+|spark.sql, bigquery, cassandra, elasticsearch, file, hbase, ignite, jdbc, kylin, livy, markdown, neo4j, pig, python, shell, zengine    |
```

Comment:
listing this is quite error prone - this is too easy to forget to update. 
instead of the black list, how about we put the white list instead? ie. the ones that have object interpolation? is that shorter?

Reply:
This is in fact the white-list - but for 2 features: dynamic-forms and object-interpolation. 
The interpreters that implement object-interpolation now have a _(*)_ suffix which is explained in text below the table.